packages/frontend/@n8n/chat/src/components/Chat.vue

```diff
@@ -42,11 +52,76 @@ function closeChat() {
 	chatEventBus.emit('close');
 }
 
+function onArrowKeyDown(payload: ArrowKeyDownPayload) {
+	const userMessagesList = userMessages.value;
+
+	if (userMessagesList.length === 0) {
+		return;
+	}
+
+	// Save current input if we're starting navigation
+	if (messageHistoryIndex.value === -1 && payload.currentInputValue.length > 0) {
+		currentInputBuffer.value = payload.currentInputValue;
+	}
+
+	if (payload.key === 'ArrowUp') {
+		// Temporarily blur to avoid cursor position issues
+		chatEventBus.emit('blurInput');
+
+		// Navigate to previous message
+		if (messageHistoryIndex.value < userMessagesList.length - 1) {
+			messageHistoryIndex.value++;
+			const messageText = userMessagesList[userMessagesList.length - 1 - messageHistoryIndex.value];
+			chatEventBus.emit('setInputValue', messageText);
+		}
+
+		// Refocus and move cursor to end
+		chatEventBus.emit('focusInput');
+	} else if (payload.key === 'ArrowDown') {
+		// Only navigate if we're in history mode
+		if (messageHistoryIndex.value === -1) return;
+
+		// Temporarily blur to avoid cursor position issues
+		chatEventBus.emit('blurInput');
+
+		// Navigate to next message or restore original input
+		if (messageHistoryIndex.value > 0) {
+			messageHistoryIndex.value--;
+			const messageText = userMessagesList[userMessagesList.length - 1 - messageHistoryIndex.value];
+			chatEventBus.emit('setInputValue', messageText);
+		} else if (messageHistoryIndex.value === 0) {
+			// Reached the end - restore original input or clear
+			messageHistoryIndex.value = -1;
+			chatEventBus.emit('setInputValue', currentInputBuffer.value);
+			currentInputBuffer.value = '';
+		}
+
+		// Refocus and move cursor to end
+		chatEventBus.emit('focusInput');
+	}
+}
+
+function onEscapeKeyDown() {
```

Comment:
Trying to test this, but Esc does not seem to be having any effect. Can you explain what this is supposed to do? Or how to test it?